util/fipstools/inject_hash/macho_parser/tests/macho_tests.h

```diff
@@ -0,0 +1,278 @@
+        uint32_t index = 0;
+
+        while (*symbol != '\0') {
+            if (strcmp(symbol, symbol_name) == 0) {
+                return index;
+            }
+
+            index += strlen(symbol) + 1;
+            symbol += strlen(symbol) + 1;
+        }
+
+        return UINT32_MAX;
+    }
+
+    static void SetUpTestSuite() {
+        bool fail = true;
+        section_info *expected_text_section = NULL;
+        section_info *expected_const_section = NULL;
+        section_info *expected_symbol_table = NULL;
+        section_info *expected_string_table = NULL;
+        section_info *expected_sections = NULL;
+
+        struct nlist_64 symbol1;
+        struct nlist_64 symbol2;
+
+        static FILE *file = fopen(TEST_FILE, "wb");
+        if (file == NULL) {
+            LOG_ERROR("Error with fopen() on %s file", TEST_FILE);
+        }
+
+        uint32_t header_sizeofcmds = sizeof(struct segment_command_64) + 2 * sizeof(struct section_64) + sizeof(struct symtab_command);
+        uint32_t header_ncmds = 2;
+        struct mach_header_64 test_header = {
+            .magic = MH_MAGIC_64,
+            .ncmds = header_ncmds,
+            .sizeofcmds = header_sizeofcmds,
+        };
+
+        uint32_t text_segment_cmdsize = sizeof(struct segment_command_64) + 2 * sizeof(struct section_64);
+        uint32_t text_segment_nsects = 2;
+        struct segment_command_64 test_text_segment = {
+            .cmd = LC_SEGMENT_64,
+            .cmdsize = text_segment_cmdsize,
+            .segname = "__TEXT",
+            .nsects = text_segment_nsects,
+        };
+
+        uint32_t text_section_offset = sizeof(struct mach_header_64) + sizeof(struct segment_command_64) + 2 * sizeof(struct section_64) + sizeof(struct symtab_command);
+        uint64_t text_section_size = TEXT_DATA_SIZE; // {0xC3}
+        struct section_64 test_text_section = {
+            .sectname = "__text",
+            .size = text_section_size, 
+            .offset = text_section_offset,
+        };
+
+        uint32_t const_section_offset = text_section_offset + text_section_size;
+        uint64_t const_section_size = CONST_DATA_SIZE;  // "hi"
+        struct section_64 test_const_section = {
+            .sectname = "__const",
+            .size = const_section_size,
+            .offset = const_section_offset,
+        };
+
+        uint32_t symtab_command_symoff = const_section_offset + const_section_size;
+        uint32_t symtab_command_stroff = symtab_command_symoff + NUM_SYMS * sizeof(struct nlist_64);
+        uint32_t symtab_command_strsize = 32;
+        struct symtab_command test_symtab_command = {
+            .cmd = LC_SYMTAB,
+            .cmdsize = sizeof(struct symtab_command),
+            .symoff = symtab_command_symoff,
+            .nsyms = NUM_SYMS,
+            .stroff = symtab_command_stroff,
+            .strsize = symtab_command_strsize,
+        };
+
+        if (fwrite(&test_header, sizeof(struct mach_header_64), 1, file) != 1) {
```

Comment:
I dunno what the definition are of these structures. But generally `sizeof(struct) != <sum of size of fields in struct>`.

This should work locally here, because you use sizeof(struct) to be both write and read. But in general, this is not the correct way to serialise.